### PR TITLE
[WIP] Move from HSV to HSL

### LINKED ISSUE #98: Switch to HSL from HSV
[hsl is standard CSS color format, while hsv is not](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value). This color picker should use the hsl color model. This is currently causing upstream issues for some other Mapbox projects.

### PR DESCRIPTION
**Pull request by tristen**
- Changes that really require review is how each gradient shapes the color space. This changed a bit as the color space incorporates 50% grey (I found this SO question helpful https://stackoverflow.com/a/31322679/260457)
- Gradients are now nested under `<XYController />` so absolute positioning within its relative container is met.
- HSL is now the default mode which makes it a bit more clear which channel is selected by default (`h`).
- A lot of the changes are to tests which went through some renaming and snapshot revisions
- `opacityLow` and `opacityHigh` is now passed as a number so they can be handled differently between each gradient component
- Not many changes were required of any of the color functions. The color utils out of the box support HSL!
- Small change outside of the intent: Adjusting the alpha slider should have an effect on the isDark state of the XYcontroller so I'm passing a value without the alpha channel here https://github.com/mapbox/react-colorpickr/pull/110/files#diff-551d240a2ff413ed0206ffd80918693fR367.

---

Closes #98

### CODE REVIEW

**Comment by davidtheclark** [comment]
Looks good to me. I don't know what I'm doing, though, with "color spaces" and what not. Code looks good. 

The only problem I see is that you can't delete a `0` in one of the channel inputs. It looks like that's an existing problem, not a new one, though.

File: test/jest/__snapshots__/l-gradient.jest.js.snap [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`LGradient basic renders 1`] = `<noscript />`;
+
+exports[`LGradient opacity renders 1`] = `<noscript />`;
```
Comment (dasulit) [inline test/jest/__snapshots__/l-gradient.jest.js.snap]: These don't seem super informative?

**Review by dasulit — APPROVED** [review]
@tristen I can't provide super precise feedback on the color function changes, but poking around things are looking good to me. I like your decision to use HSL as the default 👍

File: test/jest/__snapshots__/l-gradient.jest.js.snap (hunk 1 above)
Comment (tristen) [reply test/jest/__snapshots__/l-gradient.jest.js.snap]: Ahh yeah ... at least one of these test cases should have an active prop passed. Thanks for the catch

**Pull request merged**